consensus/state.go

```diff
@@ -1890,7 +1905,15 @@ func (cs *State) addProposalBlockPart(msg *BlockPartMessage, peerID p2p.ID) (add
 			return added, err
 		}
 
+		data, err := cs.blockExec.HydrateBlockData(context.TODO(), block)
+		if err != nil {
+			return added, err
+		}
+		block.Data = data
```

Comment:
there is two parts two these changes. 

1. the block part set we communicate to peers is now a separate part set than what is used for storage/voting/etc. We use the consensus part set to establish communications and...
2. When necessary, hydrate any of the missing txs (aka, hydrate narwhal collections into txs) here once we have completed receiving the consensus block part set.